src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala

```diff
@@ -237,6 +237,32 @@ class Initializer(config: Config) extends SparkSessionWrapper {
     config.setTempWorkingDir(workspaceTempWorkingDir)
   }
 
+  private[overwatch] def validateApiEnv(apiEnv: ApiEnv) = {
+    if (apiEnv.threadPoolSize < 1 || apiEnv.threadPoolSize > 20) {
+      throw new BadConfigException("ThreadPoolSize should be a valid number between 0 to 20")
+    }
+    if (apiEnv.apiWaitingTime < 60000 || apiEnv.apiWaitingTime > 900000) { // 60000ms = 1 mint,900000ms = 15mint
+      throw new BadConfigException("ApiWaiting time should be between 60000ms and 900000ms")
+    }
+    if (apiEnv.errorBatchSize < 1 || apiEnv.errorBatchSize > 1000) {
+      throw new BadConfigException("ErrorBatchSize should be between 1 to 1000")
+    }
+    if (apiEnv.successBatchSize < 1 || apiEnv.successBatchSize > 1000) {
+      throw new BadConfigException("SuccessBatchSize should be between 1 to 1000")
+    }
+    if (apiEnv.proxyHost.isDefined) {
```

Comment:
`!isDefined` should always be `isEmpty` isDefined is for nulls, these are Options

Switch 
`isDefined` to `.nonEmpty` and
`!isDefined` to `.isEmpty` 

Like shown below. 

`apiEnv.proxyHost.isDefined` --> `apiEnv.proxyPort.nonEmpty`
`!apiEnv.proxyPort.isDefined` --> `apiEnv.proxyPort.isEmpty`

please do this everywhere an option type is referenced and you've used `isDefined`